azavea/opendataphilly-ckan
# Remove unused extensions

## Overview

Removes some CKAN extensions that were unused or under-used and not worth maintaining.
- `deadoralive`: Already disabled, in PR #70, due to false-alarm problems. This removes the inactive provisioning roles.
- `harvest`: Installed but not added to the plugins list in the config, and the database tables were apparently never initialized.
- `archiver`: Jobs were being triggered to archive datasets, but the destination was not configured and no related functionality was exposed in the front-end.
- `issues`: The plugin was installed and active, but no links were offered to create or view issues in the front-end, so none were ever created.  Since this would not be our preferred feedback mechanism anyway, removing it makes sense.

### Notes

Some of these extensions come with database tables.  `archiver` doesn't, at least not in the version we had installed, and since `harvester` was never initialized its tables are not in the database.  `deadoralive` created `link_checker_results` and `issues` created `issue_category`, `issue_comment`, and `issue`.  We should delete those tables once the extensions are gone.  There's no harm in having them still there but ignored, so there's no tricky timing considerations--we just need to wait until the version without the extensions is launched and stable, then we can remove them at any time.

There are ways to make data migrations in CKAN, and ways to add paster commands to extensions that could be used for migration, but there's not really any place to put a migration related to the removal of an extension.  So I haven't thought of any better plan than getting a DB shell and running:
```sql
BEGIN;

-- Table for 'deadoralive'
DROP TABLE link_checker_results;

-- Tables for 'issues'
DROP TABLE issue_category;
DROP TABLE issue_comment;
DROP TABLE issue;

COMMIT;
```

## Testing Instructions

- Remove the `ckan_deadoralive_apikey` line from your `deployment/ansible/group_vars/vagrant` file (might not actually be necessary, but better to keep in sync).
- Destroy your development VM and provision from scratch.
- Since this is mostly removing unused functionality, there's not much to see.  There should be no errors during provisioning and the site should work normally.
